tableone.py

```diff
@@ -98,7 +100,11 @@ def __init__(self, data, columns=None, categorical=None, groupby=None,
             raise InputError('Input contains duplicate columns: {}'.format(dups))
 
         # if categorical not specified, try to identify categorical
-        if not categorical and type(categorical) != list:
+        if type(columns) == type(None):
+            categorical = self._detect_categorical_columns(data[columns])
+        elif 'pandas.core.indexes' in str(type(categorical)):
+            categorical = categorical.get_values()
+        elif type(categorical) != list:
```

Comment:
If someone specifies a single categorical variable as a string (e.g. `categorical = 'ICU'`), then I think we now ignore their selection and generate the list ourselves. Might be good to catch the string too?